tests/test_helper.tcl

```diff
@@ -600,6 +601,9 @@ proc print_help_screen {} {
         "--tls-module       Run tests in TLS mode with Valkey module."
         "--host <addr>      Run tests against an external host."
         "--port <port>      TCP port to use against external host."
+        "--old-server-path <path>"
```

Comment:
IMHO using `--old-server-path` might limit the scope to only older versions, while `--other-server-path` or `--compat-server-path` feels less restrictive and better reflects its purpose for inter-version compatibility testing.

Reply:
I thought about words like "other", "alternative" server, etc. This is how I thought when I picked "old":

* If we're testing the compatibility between an older and a newer versions, let's say Valkey 8.1 and Valkey 9.0, either upgrade, downgrade or cluster bus compatibility, those are tests that we would add in the later version, Valkey 9 in this case. We wouldn't normally add tests in an old version to test against a future version.
* A name like "other server" is less clear when reading a test case about compatibility between and old and a new version. If you see "old server", it's very clear which one is the new version and which is the old version. A name like other would be more ambiguous.

Not sure the above are important though and I understand your point that we can also test against something like KeyDB, AWS MemoryDB, Rockskv, etc. Also the same Valkey version compiled in different ways, with/without some modules, without Lua, etc, etc. Yes, there are definitely more use cases than old and new, so I agree something like "other" is more generic.

Other just means *not the system under test*, which is what matters here. "Compat" doesn't seem general enough though. We may (hypothetically) want to test something against some *incompatible* version to test the failure modes (e.g. that we log some error and don't crash).

I'll change to "other" then, unless you have more ideas or pros/cons.

Reply:
I proposed secondary in my other comment, I like other too. I don't really like `compat`.

Reply:
I've head secondary in the context of primary-secondary, meaning primary-replica.

I prefer "other" so far. The essence is that it's a different one than the SUT one. Other = different.

We could use "different", but `--different-server-path` doesn't look that great.

Reply:
Changed to "old"